framework/validators/FileValidator.php

```diff
@@ -258,7 +259,12 @@ protected function validateValue($file)
      */
     public function getSizeLimit()
     {
+        // Max POST size minus some bytes for form data 
+        $post_limit = $this->sizeToBytes(ini_get('post_max_size')) - 1024;
```

Comment:
Do not forget, that form may actually contain only single file input, so reserving any value here is inconsistent.